### PR TITLE
[REFACTOR] Auto Lock section

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

Refactor 'Auto Lock' setting to TS and extract it from the SecuritySettings file.

**Screenshots/Recordings**

_Not applicable_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/Settings/SecuritySettings/Sections/AutoLock/AutoLock.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import AUTO_LOCK_OPTIONS from './constants';
+import { setLockTime } from '../../../../../../actions/settings';
+import { useTheme } from '../../../../../../util/theme';
+import SelectComponent from '../../../../../UI/SelectComponent';
+import Text from '../../../../../../component-library/components/Texts/Text';
+import { strings } from '../../../../../../../locales/i18n';
+
+const AutoLock = () => {
+  const { colors } = useTheme();
+  const styles = createStyles(colors);
```
Comment (Cal-L) [inline app/components/Views/Settings/SecuritySettings/Sections/AutoLock/AutoLock.tsx:14]: You can remove useTheme and replace createStyles with useStyles, which exposes theme for use in the style sheet

File: app/components/Views/Settings/SecuritySettings/Sections/AutoLock/styles.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { fontStyles } from '../../../../../../styles/common';
+import { StyleSheet } from 'react-native';
+
+export const createStyles = (colors: any) =>
+  StyleSheet.create({
+    setting: {
+      marginTop: 50,
+    },
+    title: {
+      ...fontStyles.normal,
```
[the remark this answers is not in the document]
Comment (Cal-L) [reply app/components/Views/Settings/SecuritySettings/Sections/AutoLock/styles.ts:11]: useStyles will also expose typography from the theme param

[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/Settings/SecuritySettings/Sections/AutoLock/styles.ts:11]: Looks like here is a typing issue in the DS with the `Text` component and `typography`. I'll leave as it is until that is fixed,

```
Type 'string' is not assignable to type '"normal" | "bold" | "100" | "200" | "300" | "400" | "500" | "600" | "700" | "800" | "900" | undefined'.
```

**Review by Cal-L — APPROVED** [review]
Lgtm

**Pull request merged**